File: java/Glacier2/session/server/src/main/java/com/example/glacier2/session/server/PokeStore.java
// Copyright (c) ZeroC, Inc.

package com.example.glacier2.session.server;

import java.util.List;

interface PokeStore {
    /**
     * Saves the Pokémon collection of a specific user.
     * @param userId the user's ID
     * @param pokemon the Pokémon collection to save
     */
    void saveCollection(String userId, List<String> pokemon);

    /**
     * Retrieves the Pokémon collection of a specific user.
     * @param userId the user's ID
     * @return the saved Pokémon collection, or {@code null} if no collection was saved for {@code userId}.
     *     The returned list is immutable. Attempting to modify it will throw an UnsupportedOperationException.
     */
    List<String> retrieveCollection(String userId);
}

File: swift/Glacier2/Session/Sources/Server/PokeStore.swift
// Copyright (c) ZeroC, Inc.

import Foundation
import Ice

/// Represents a Pokémon storage system.
protocol PokeStore {
    /// Saves the Pokémon collection for a specific user.
    /// - Parameters:
    ///   - userId: The user ID.
    ///   - pokemon: The Pokémon collection to save.
    func saveCollection(userId: String, pokemon: [String])

    /// Retrieves the Pokémon collection for a specific user.
    /// - Parameter userId: The user ID.
    /// - Returns: The saved Pokémon collection, or an empty list if no collection was saved.
    func retrieveCollection(userId: String) -> [String]
}
